raftLogCheck.js

```diff
@@ -0,0 +1,235 @@
+                            }
+                            if (leaderLog.log[i].entries[j].key === undefined) {
+                                if (leaderLog.log[i].method === 0 || // CREATE
+                                    leaderLog.log[i].method === 1 || // DELETE
+                                    leaderLog.log[i].method === 7) { // PUT_ATTR
+                                    expectKey = false;
+                                } else {
+                                    log.warning('undefined leader key at', {
+                                        seq: seq + i,
+                                        off: j,
+                                        db: leaderLog.log[i].db,
+                                        entry: leaderLog.log[i].entries[j] });
+                                    continue;
+                                }
+                            }
+                            if (leaderLog.log[i].entries[j].value ===
+                                undefined) {
+                                if (leaderLog.log[i].entries[j].type ===
+                                    'del') {
+                                    expectValue = false;
+                                } else {
+                                    log.debug('undefined leader value at', {
+                                        seq: seq + i,
+                                        off: j,
+                                        entry: leaderLog.log[i].entries[j] });
+                                    continue;
+                                }
+                            }
+                            if (memberLog.log[i].entries[j].key ===
+                                undefined) {
+                                if (expectKey) {
+                                    log.debug('undefined member key at', {
+                                        seq: seq + i,
+                                        off: j });
+                                }
+                            }
+                            if (memberLog.log[i].entries[j].value ===
+                                undefined) {
+                                if (expectValue) {
+                                    log.debug('undefined member value at', {
+                                        seq: seq + i,
+                                        off: j,
+                                        entry: memberLog.log[i].entries[j] });
+                                }
+                            }
+                            if (memberLog.log[i].entries[j].key !==
+                                leaderLog.log[i].entries[j].key) {
+                                log.debug('key mismatch at', {
+                                    seq: seq + i,
+                                    off: j,
+                                    key: memberLog.log[i].entries[j].key });
+                            }
+                            if (md5(memberLog.log[i].entries[j].value) !==
```

Comment:
You can remove the `md5()` calls as well as the dependency on `md5` module, I don't see any benefit in using md5 in this case.